sds_data_manager/lambda_code/SDSCode/api_lambdas/download_api.py

```diff
@@ -6,11 +6,58 @@
 
 import boto3
 import botocore
+import imap_data_access
+
+from ..api_lambdas.utils import is_authenticated_user
+from ..database import database as db
+from ..database import models
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
+def is_released(s3_key):
+    """Check if the file is marked as released in the database.
+
+    Parameters
+    ----------
+    s3_key : str
+        The S3 key path of the file to check.
+
+    Returns
+    -------
+    bool
+        True if the file is released, False otherwise.
+    """
+    try:
+        filename = os.path.basename(s3_key)
+        file_obj = imap_data_access.file_validation.generate_imap_file_path(filename)
+
+        # Determine which table to use based on file type
+        if hasattr(file_obj, "__class__") and hasattr(file_obj.__class__, "__name__"):
+            class_name = file_obj.__class__.__name__
+            if class_name == "SPICEFilePath":
+                table = models.SPICEFiles
+            elif class_name == "AncillaryFilePath":
+                table = models.AncillaryFiles
+            elif class_name == "ScienceFilePath":
+                table = models.ScienceFiles
+            else:
+                return False
+        else:
+            return False
+
+        with db.Session() as session:
+            file_record = session.query(table).filter(table.file_path == s3_key).first()
+            if file_record and hasattr(file_record, "release"):
+                return file_record.release
```

Comment:
Here is one with the "release" when it should be "released" I believe.

```suggestion
            if file_record:
                return getattr(file_record, "released", False)
```